microceph/ceph/keyring.go

```diff
@@ -106,3 +106,39 @@ func parseKeyring(path string) (string, error) {
 
 	return cephSecret, nil
 }
+
+// CreateKey creates a client key and returns the said key hash without saving it as a file.
+var CreateKey = func(clientName string, caps ...[]string) (string, error) {
```

Comment:
Is there a reason this is a func variable and not a regular function? And, does it need to be public?

Reply:
Public func variables can be mocked with a stub implementation in unit tests. This is not possible otherwise. This way we can mock any low level API accessing a ceph command (in this case creating a key) and provide dummy keys for verification.